packages/react/src/with-current-user.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/no-did-mount-set-state */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class WithCurrentUser extends Component {
+  constructor(props, context) {
+    super(props);
+    const user = context.accountsClient.user();
+    const loading = context.accountsClient.getState().get('isLoading');
+    this.state = {
+      user,
+      loading,
+    };
+  }
+
+  componentDidMount() {
+    const { accountsClient } = this.context;
+    this.unsubscribe = this.context.accountsClient.store.subscribe(() => {
```

Comment:
There is no way to make the subscribe callback more specific? like only try to set state if an action with user or loading was dispatched?
I think that this way we will cause a re-render for each component that uses that for every action..